src/main/java/pro/belbix/ethparser/codegen/SimpleContractGenerator.java

```diff
@@ -103,26 +120,42 @@ private boolean equalContracts(GeneratedContract newContract, GeneratedContract
       Long block,
       boolean isProxy,
       String selector,
-      String network
+      String network,
+      Boolean isUseCache
   ) {
 
     boolean isOverride = isOverrideAbi(address);
     String abi = null;
     String contractName = "UNKNOWN";
     boolean etherscanIsProxy = false;
     String etherscanProxyImpl = "";
-
-    AbiProviderService.SourceCodeResult sourceCode =
-        abiProviderService.contractSourceCode(
+    AbiProviderService.SourceCodeResult sourceCode;
+    if (isUseCache) {
+      ContractSourceCodeDTO cashed_source =
+          contractSourceCodeRepository.findByAddressNetwork(address, network);
+      if (cashed_source != null) {
+        log.info("Used cached contract sources code {}", cashed_source.getContractName());
+        sourceCode = ContractSourceModelConverter.toSourceCodeResult(cashed_source);
+      } else {
+        sourceCode = abiProviderService.contractSourceCode(
             address, networkProperties.get(network).getAbiProviderKey(), network);
+        ContractSourceCodeDTO csdto = ContractSourceModelConverter.toDTO(sourceCode);
+        csdto.setAddress(address);
+        csdto.setNetwork(network);
+        contractSourceCodeRepository.save(csdto);
+      }
+    } else {
+      sourceCode = abiProviderService.contractSourceCode(
```

Comment:
duplicate code, we can use simplier if statement

Reply:
Ack